### ISSUE TITLE
Build Genome Database stuck on ebwt loop

### ISSUE THREAD

**Issue by astephen333** [ENCODE-DCC/atac-seq-pipeline#390]
## **Describe the bug**

Tried to build rheMac10 genome with an sbatch. Ran "sbatch --mem=16G -p pfen2 -o rhemac10.o -J rhemac10 -e rhemac10err.sh /home/azstephe/atac-seq-pipeline/rhemac10.sh --pty/bin/bash"

This is the rhemac10.sh file.
#!/bin/bash
source /home/azstephe/miniconda3/etc/profile.d/conda.sh
conda activate encd-atac
bash /home/azstephe/atac-seq-pipeline/scripts/build_genome_data.sh macaca /projects/pfenninggroup/machineLearningForComputationalBiology/immunetacit/genomes/rhemac

This is the output file, rhemac10.o. Entered the Ebwt loop a few hours after being sbatched and has been stuck on the Ebwt loop for 7 days.
=== Creating destination directory and TSV file...
=== Downloading files...
=== Processing reference fasta file...
=== Generating fasta index and chrom.sizes file...
=== Extracting mito chromosome from fasta
=== Determinig gensz...
=== Building bowtie2 index...
Settings:
  Output files: "rheMac10.fa.*.bt2"
  Line rate: 6 (line is 64 bytes)
  Lines per side: 1 (side is 64 bytes)
  Offset rate: 4 (one in 16)
  FTable chars: 10
  Strings: unpacked
  Max bucket size: default
  Max bucket size, sqrt multiplier: default
  Max bucket size, len divisor: 8
  Difference-cover sample period: 1024
  Endianness: little
  Actual local endianness: little
  Sanity checking: disabled
  Assertions: disabled
  Random seed: 0
  Sizeofs: void*:8, int:4, long:8, size_t:8
Input files DNA, FASTA:
  rheMac10.fa
Reading reference sizes
  Time reading reference sizes: 00:00:20
Calculating joined length
Writing header
Reserving space for joined string
Joining reference sequences
  Time to join reference sequences: 00:00:14
bmax according to bmaxDivN setting: 367111590
Using parameters --bmax 275333693 --dcv 1024
  Doing ahead-of-time memory usage test
  Passed!  Constructing with these parameters: --bmax 275333693 --dcv 1024
Constructing suffix-array element generator
Building DifferenceCoverSample
  Building sPrime
  Building sPrimeOrder
  V-Sorting samples
  V-Sorting samples time: 00:01:00
  Allocating rank array
  Ranking v-sort output
  Ranking v-sort output time: 00:00:15
  Invoking Larsson-Sadakane on ranks
  Invoking Larsson-Sadakane on ranks time: 00:00:30
  Sanity-checking and returning
Building samples
Reserving space for 22 sample suffixes
Generating random suffixes
QSorting 22 sample offsets, eliminating duplicates
QSorting sample offsets, eliminating duplicates time: 00:00:00
Multikey QSorting 22 samples
  (Using difference cover)
  Multikey QSorting samples time: 00:00:00
Calculating bucket sizes
Splitting and merging
  Splitting and merging time: 00:00:00
Split 3, merged 11; iterating...
Splitting and merging
  Splitting and merging time: 00:00:00
Split 3, merged 2; iterating...
Splitting and merging
  Splitting and merging time: 00:00:01
Avg bucket size: 2.09778e+08 (target: 275333692)
Converting suffix-array elements to index image
Allocating ftab, absorbFtab
Entering Ebwt loop 

## **OS/Platform**
- OS/Platform: Linux cluster centos 7 with slurm.
- Conda version: conda 4.10.1
- Pipeline version: v2.2.0
- Caper version: v2.2.0

## **Caper configuration file**
backend=slurm
```ini
slurm-partition=pfen2
```

## **Input JSON file**
Paste contents of your input JSON file.
```json
PASTE INPUT JSON CONTENTS HERE
```

## **Troubleshooting result**

If you ran `caper run` without Caper server then Caper automatically runs a troubleshooter for failed workflows. Find troubleshooting result in the bottom of Caper's screen log.

If you ran `caper submit` with a running Caper server then first find your workflow ID (1st column) with `caper list` and run `caper debug [WORKFLOW_ID]`.

Paste troubleshooting result.
```
PASTE TROUBLESHOOTING RESULT HERE
```

**Comment by imk1** [comment]
We have now had this issue with multiple versions of the pipeline and of anaconda.  The problem also seems to happen when making the mitochondrial genome index.  It would be great if someone could help us resolve this.

**Comment by imk1** [comment]
I think that we solved the problem.  This seems to be a problem with bowtie2 version 2.3.4.3.  We first upgraded libgcc-ng to version 12, which we were able to get through the conda-forge channel; this version of libgcc-ng seems to be required for newer versions of bowtie2.  We then upgraded bowtie2 from version 2.3.4.3 to version 2.5.4, which we were able to get through the bioconda channel.  I recommend changing the dependencies of the pipeline to a different version of bowtie2.
